diff --git a/pkgs/by-name/my/mygui/package.nix b/pkgs/by-name/my/mygui/package.nix
index d574bbc73e1f..40346b974533 100644
--- a/pkgs/by-name/my/mygui/package.nix
+++ b/pkgs/by-name/my/mygui/package.nix
@@ -20,17 +20,17 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "mygui";
-  version = "3.4.2";
+  version = "3.4.3";
 
   src = fetchFromGitHub {
     owner = "MyGUI";
     repo = "mygui";
     rev = "MyGUI${version}";
-    hash = "sha256-yBV0ImOFJlqBPqqOjXYe4SFO2liSGZCEwvehED5Ubj4=";
+    hash = "sha256-qif9trHgtWpYiDVXY3cjRsXypjjjgStX8tSWCnXhXlk=";
   };
 
   patches = [
-    ./disable-framework.patch
+    # ./disable-framework.patch
   ];
 
   nativeBuildInputs = [
@@ -61,6 +61,7 @@ stdenv.mkDerivation rec {
     "-DMYGUI_BUILD_TOOLS=OFF"
     "-DMYGUI_BUILD_DEMOS=OFF"
     "-DMYGUI_RENDERSYSTEM=${renderSystem}"
+    "-DMYGUI_DONT_USE_OBSOLETE=ON"
   ];
 
   meta = with lib; {
diff --git a/pkgs/games/openmw/default.nix b/pkgs/games/openmw/default.nix
index cff3576ebafe..f9bcf0fbfb28 100644
--- a/pkgs/games/openmw/default.nix
+++ b/pkgs/games/openmw/default.nix
@@ -5,7 +5,8 @@
   fetchpatch,
   cmake,
   pkg-config,
-  wrapQtAppsHook,
+  qt6, # wrapQtAppsHook,
+  kdePackages,
   SDL2,
   boost,
   bullet,
@@ -66,16 +67,16 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "openmw";
-  version = "0.48.0";
+  version = "49-rc6";
 
   src = fetchFromGitLab {
     owner = "OpenMW";
     repo = "openmw";
     rev = "${pname}-${version}";
-    hash = "sha256-zkjVt3GfQZsFXl2Ht3lCuQtDMYQWxhdFO4aGSb3rsyo=";
+    hash = "sha256-ZpZycprNNx2MDmmWNwCixIG6va5e1EvRjxnWbjIXzaE=";
   };
 
-  patches = [ ./0001-function-inclusion-fixes-for-gcc14.patch ];
+  # patches = [ ./0001-function-inclusion-fixes-for-gcc14.patch ];
 
   postPatch =
     ''
@@ -89,13 +90,15 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     cmake
     pkg-config
-    wrapQtAppsHook
+    qt6.wrapQtAppsHook
   ];
 
   # If not set, OSG plugin .so files become shell scripts on Darwin.
   dontWrapQtApps = stdenv.hostPlatform.isDarwin;
 
   buildInputs = [
+    qt6.qtbase
+    kdePackages.qttools
     SDL2
     boost
     bullet'
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index ff8f4ec6938d..2fb66c005ab4 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15133,7 +15133,7 @@ with pkgs;
 
   openloco = pkgsi686Linux.callPackage ../games/openloco { };
 
-  openmw = libsForQt5.callPackage ../games/openmw { };
+  openmw = callPackage ../games/openmw { };
 
   openmw-tes3mp = libsForQt5.callPackage ../games/openmw/tes3mp.nix { };
 
